connectors/wsclient/src/main/java/quarks/connectors/wsclient/runtime/WebSocketClientReceiver.java

```diff
@@ -0,0 +1,45 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2015,2016 
+*/
+package quarks.connectors.wsclient.runtime;
+
+import quarks.function.Consumer;
+import quarks.function.Function;
+
+public class WebSocketClientReceiver<T> implements Consumer<Consumer<T>>, AutoCloseable {
+    private static final long serialVersionUID = 1L;
+    protected final WebSocketClientConnector connector;
+    private final Function<String,T> toTuple;
+    protected Consumer<T> eventHandler;
+    
+    public WebSocketClientReceiver(WebSocketClientConnector connector, Function<String,T> toTuple) {
+        this.connector = connector;
+        this.toTuple = toTuple;
+    }
+
+    @Override
+    public void accept(Consumer<T> eventHandler) {
+        this.eventHandler = eventHandler;
+        connector.setReceiver(this);
```

Comment:
Could `connector.setReceiver(this)` be called once in the constructor, instead of on every `accept()`?